src/rustbox.rs

```diff
@@ -24,27 +24,33 @@ use libc::c_int;
 use gag::Hold;
 
 pub mod keyboard;
+pub mod mouse;
 
 pub use keyboard::Key;
+pub use mouse::Mouse;
 
 #[derive(Clone, Copy)]
 pub enum Event {
     KeyEventRaw(u8, u16, u32),
     KeyEvent(Option<Key>),
     ResizeEvent(i32, i32),
+    MouseEvent(Mouse, i32, i32),
     NoEvent
 }
 
-#[derive(Clone, Copy, Debug)]
-pub enum InputMode {
-    Current = 0x00,
-
-    /// When ESC sequence is in the buffer and it doesn't match any known
-    /// ESC sequence => ESC means TB_KEY_ESC
-    Esc     = 0x01,
-    /// When ESC sequence is in the buffer and it doesn't match any known
-    /// sequence => ESC enables TB_MOD_ALT modifier for the next keyboard event.
-    Alt     = 0x02,
+bitflags! {
```

Comment:
Ok, cool. Just hit that assertion, prob something that should be fixed in termbox, but anyways.